CHAID/CHAID.py

```diff
@@ -276,10 +276,14 @@ def __init__(self, ndarr, arr, alpha_merge=0.05, max_depth=2, min_sample=30, spl
             self.vectorised_array.append(CHAIDVector(ndarr[:, ind]))
         self.data_size = ndarr.shape[0]
         self.node_count = 0
-        self.tree_store = []
-        self.observed = arr
+        self.tree_store = None
+        self.observed = CHAIDVector(arr)
```

Comment:
Regardless we need to do a deep copy, this was causing some of the tests to fail.